solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -541,26 +570,127 @@ contract RandomBeacon is IRandomBeacon, Ownable {
         sortitionPool.withdrawIneligible(recipient);
     }
 
-    /// @notice Registers the caller in the sortition pool.
-    function registerOperator() external {
-        address operator = msg.sender;
+    /// @notice Used by staking provider to set operator address that will
+    ///         operate a node. The given staking provider can set operator
+    ///         address only one time. The operator address can not be changed
+    ///         and must be unique. Reverts if the operator is already set for
+    ///         the staking provider or if the operator address is already in
+    ///         use. Reverts if there is a pending authorization decrease for
+    ///         the staking provider.
+    function registerOperator(address operator) external {
+        authorization.registerOperator(operator);
+    }
 
-        require(
-            !sortitionPool.isOperatorInPool(operator),
-            "Operator is already registered"
+    /// @notice Lets the operator join the sortition pool. The operator address
+    ///         must be known - before calling this function, it has to be
+    ///         appointed by the staking provider by calling `registerOperator`.
+    ///         Also, the operator must have the minimum authorization required
+    ///         by the application. Function reverts if there is no minimum stake
```

Comment:
I wanted the docs to be more generic, in case we want to copy-paste it to another application. But if you think it's better the way you proposed I'll update it.

Reply:
472fcdf03dd3093473cd36d98fbbc9433276f2ca

Reply:
Yeah, I think it makes sense to be less-generic in docs. Can we also make changes in `RandomBeacon.sol` similar to the ones from https://github.com/keep-network/keep-core/commit/472fcdf03dd3093473cd36d98fbbc9433276f2ca ?

Reply:
Oh, yeah, I missed it, thanks for noticing that! Updated in cf8dc122d